sdk/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics.export;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.util.concurrent.MoreExecutors;
+import io.opentelemetry.internal.Utils;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import jdk.nashorn.internal.ir.annotations.Immutable;
+
+/**
+ * Wraps a list of {@link MetricProducer}s and automatically reads and exports the metrics every
+ * export interval.
+ *
+ * @since 0.3.0
+ */
+public final class IntervalMetricReader {
+  private final Exporter exporter;
+  private final ScheduledExecutorService scheduler;
+
+  /**
+   * Stops the scheduled task and calls export one more time.
+   *
+   * @since 0.3.0
+   */
+  public void shutdown() {
+    scheduler.shutdown();
+    exporter.run();
+  }
+
```

Comment:
We said that we need to wait for the in-progress task if any.

Reply:
I don't think we ever came to a conclusion on how we wanted to behave, but I'm happy to have this method block until everything is done, if that's what is desired.

Reply:
How long should we wait before bailing out, do you think?

Reply:
I made it wait 5 seconds. Happy to tweak it if there's a better number.